### PR TITLE
[Merged by Bors] - chore: simplify some lambdas

### PR DESCRIPTION
**Pull request by mo271**
Replaces suitable lambdas of the form

`fun f => f x` by `f` and

`fun x => x` by `id`.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Analysis/Calculus/Deriv/Mul.lean [hunk 1]
```diff
@@ -169,7 +169,7 @@ lemma deriv_const_smul' {f : 𝕜 → F} {x : 𝕜} {R : Type*} [Field R] [Modul
     · simp only [zero_smul, deriv_const']
     · have H : ¬DifferentiableAt 𝕜 (fun y ↦ c • f y) x := by
         contrapose! hf
-        change DifferentiableAt 𝕜 (fun y ↦ f y) x
+        change DifferentiableAt 𝕜 f x
```
Comment (Ruben-VandeVelde) [inline Mathlib/Analysis/Calculus/Deriv/Mul.lean:172]: Does this still do anything now? If not, remove `change` entirely.

Comment (mo271) [reply Mathlib/Analysis/Calculus/Deriv/Mul.lean]: Good point, removed!

**Review by loefflerd — COMMENTED** [review]
Definitely a step in the right direction

maintainer merge

**Comment by urkud** [comment]
Thanks! :tada:
bors merge

**Pull request closed without merging**